packages/syft/src/syft/service/code/user_code.py

```diff
@@ -675,11 +689,17 @@ def process_code(
     f = tree.body[0]
     f.decorator_list = []
 
-    keywords = [ast.keyword(arg=i, value=[ast.Name(id=i)]) for i in input_kwargs]
+    call_args = function_input_kwargs
+    # call_args = policy_input_kwargs
+    if "domain" in function_input_kwargs:
+        # print("adding domain")
+        # call_args = ["domain"] + call_args
+        context.output["uses_domain"] = True
+    call_stmt_keywords = [ast.keyword(arg=i, value=[ast.Name(id=i)]) for i in call_args]
     call_stmt = ast.Assign(
```

Comment:
this slightly changed because we (optionally) added `domain` as an argument to usercode. In general I think this code could be more clear if we refactor it a bit